pkg/operator/resource_reconciler.go

```diff
@@ -295,10 +312,17 @@ func (rr *ResourceReconciler) OnDelete(obj interface{}) {
 		return
 	}
 
-	level.Debug(rr.logger).Log("msg", fmt.Sprintf("%s deleted", rr.resourceKind), "key", key)
-	rr.metrics.TriggerByCounter(rr.resourceKind, DeleteEvent).Inc()
+	objMeta, err := meta.Accessor(obj)
+	if err != nil {
+		return
+	}
 
-	rr.reconcileQ.Add(key)
+	if rr.hasOwnership(objMeta) {
```

Comment:
(nit) Let keep the same flow and exit if the condition isn't met, it also helps to review ;)
```suggestion
	if rr.hasOwnership(objMeta) {
		return
	}
```

Reply:
done